src/game/aggregation.rs

```diff
@@ -0,0 +1,571 @@
+///     river_bet_sizes: [bet_sizes.clone(), bet_sizes],
+///     turn_donk_sizes: None,
+///     river_donk_sizes: None,
+///     add_allin_threshold: 1.5,
+///     force_allin_threshold: 0.15,
+///     merging_threshold: 0.1,
+/// };
+///
+/// let all_lines = generate_all_lines(tree_config).unwrap();
+/// assert!(!all_lines.contains(&vec![]));
+/// assert!(!all_lines.contains(&vec![Action::Check]));
+/// assert!(all_lines.contains(&vec![Action::Bet(20), Action::Fold]));
+/// assert!(all_lines.contains(&vec![Action::Bet(20), Action::Call, Action::Check, Action::Check]));
+/// ```
+pub fn generate_all_lines(config: TreeConfig) -> Result<Vec<Vec<Action>>, String> {
+    let mut action_tree = ActionTree::new(config)
+        .map_err(|e| format!("Error constructing ActionTree with input TreeConfig: {e}"))?;
+
+    let mut all_lines = Vec::new();
+    generate_all_lines_rec(&mut action_tree, &mut all_lines)?;
+
+    Ok(all_lines)
+}
+
+fn generate_all_lines_rec(
+    action_tree: &mut ActionTree,
+    lines: &mut Vec<Vec<Action>>,
+) -> Result<(), String> {
+    // NOTE: history does not include chance nodes
+    let history = action_tree.history().to_owned();
+
+    if action_tree.is_terminal_node() {
+        lines.push(history);
+        return Ok(());
+    }
+
+    let all_actions = action_tree.available_actions().to_owned();
+
+    for action in all_actions {
+        action_tree.play(action)?;
+
+        generate_all_lines_rec(action_tree, lines)?;
+
+        action_tree.back_to_root();
+        action_tree.apply_history(&history)?;
+    }
+
+    Ok(())
+}
+
+/// Tree structure for computing aggregate reports.
+/// The strucutre mirrors the action tree of the pertinent game.
+pub struct AggActionTree {
+    /// List of previous actions from this node.
+    prev_actions: Vec<Action>,
+    /// List of all available actions.
+    available_actions: Vec<Action>,
+    // NOTE: |child_trees| <= |available_actions|
+    // Because no child is created for terminating nodes
+    // Use non_terminating_available_actions to get corresponding list
+    // of actions taken to reach child trees
+    /// Map of non-terminating available actions to child `AggActionTree`s
+    child_trees: HashMap<Action, AggActionTree>,
+    /// Aggregate report rows associated with this node, one for each board at this state in the report.
+    data: Vec<AggRow>,
+}
+
+impl AggActionTree {
+    /// Initialize the AggActionTree with a set of lines and a specific TreeConfig.
+    /// To generate a report for all possible lines, first call `generate_all_lines` with the config.
+    ///
+    /// Any call to [`update_report_for_game`] on the returned `AggActionTree` should use a game
+    /// that was solved using the same `config` that was passed into `init_root`.
+    ///
+    /// [`update_report_for_game`]: #method.update_report_for_game
+    pub fn init_root(lines: Vec<Vec<Action>>, config: TreeConfig) -> Result<Self, String> {
```

Comment:
@bkushigian This is the initialization function for `AggActtionTree`. It takes the set of lines that the aggregation report should run over, and constructs the tree (w/o any data) for only those lines. Then, when `update_report_for_game` is called, only data relevant for those lines is stored.